### PR TITLE
enh: run slack webhooks on all our slack_configurations' data sources

### PR DESCRIPTION
**Pull request by spolu**
This PR starts a workflow for each SlackConfigurations' connectorIds when we receive a Slack webhook.

Otherwise if we have multiple manaaged data sources from multiple workspace pointing to the same slack team we would not update all of them (just one).

Note: we do receive only one webhook even if we connected from multiple data sources, it's because it's one webhook per Slack app per slack team.

r? @lasryaric

### CODE REVIEW

**Comment by spolu** [comment]
@lasryaric I refactored the code a bit trying to make it more readable and standardizing on the use of apiError. So please review carefully :pray:

**Comment by spolu** [comment]
Invariants I tried to preserve:
- all errors through apiError
- all 200 are preceded by a logger.info

If a workflow start fails for one we return an error which will retry the webhook but this is generally unlikely to fail.

File: connectors/src/api/webhooks/webhook_slack.ts [hunk 1]
```diff
@@ -45,55 +46,110 @@ const _webhookSlackAPIHandler = async (
...
-    if (!slackConfiguration) {
-      return res.status(404).send({
-        error: {
+    if (slackConfigurations.length === 0) {
+      return apiError(req, res, {
+        api_error: {
+          type: "connector_configuration_not_found",
           message: `Slack configuration not found for teamId ${req.body.team_id}`,
         },
+        status_code: 404,
```
Comment (lasryaric) [inline connectors/src/api/webhooks/webhook_slack.ts:72]: I am actually not sure we want to 404 here. If we don't have it and Slack tells us about it, should we just tell Slack to stop pinging us? Up to you.

File: connectors/src/api/webhooks/webhook_slack.ts [hunk 2]
```diff
@@ -45,55 +46,110 @@ const _webhookSlackAPIHandler = async (
...
-          workflowRes = await launchSlackSyncOneThreadWorkflow(
-            slackConfiguration.connectorId.toString(),
-            req.body.event.channel,
-            req.body.event.thread_ts
+    switch (req.body.event?.type) {
+      /**
+       * `message` handler.
+       */
+      case "message": {
+        if (req.body.team_id) {
```
Comment (lasryaric) [inline connectors/src/api/webhooks/webhook_slack.ts:81]: Wrong test case, you should do `if (!req.body.team_id)`.
We are going to have to test this before pushing it to prod 💯

Comment (spolu) [reply connectors/src/api/webhooks/webhook_slack.ts:81]: Nice catch

File: connectors/src/api/webhooks/webhook_slack.ts [hunk 3]
```diff
@@ -45,55 +46,110 @@ const _webhookSlackAPIHandler = async (
     return res.status(200).send({
       challenge: req.body.challenge,
     });
-  } else if (req.body.type === "event_callback") {
+  }
+
+  if (req.body.type === "event_callback") {
     if (!req.body.team_id) {
-      return res.status(400).send({
```
[the remark this answers is not in the document]
Comment (spolu) [reply connectors/src/api/webhooks/webhook_slack.ts:50]: Wilco

**Review by lasryaric — CHANGES_REQUESTED** [review]
Overall looks good. Absolutely need to be tested before going to production IMO.

**Comment by spolu** [comment]
@lasryaric what would be the tests you suggest?

Connect in dev with ngrok test that we get the updates, leaving a channel / joning a channel trigger the workflow ?

**Comment by spolu** [comment]
@lasryaric updated r? PTAL

**Review by lasryaric — APPROVED** [review]
Approved. To be tested locally or baby sited in production.

**Comment by spolu** [comment]
@lasryaric tested in dev with Dust staging app. Confirmed `channel_join` and `message` work as expected

channel_left or channel_deleted I could not test because I don't believe we've registered for these events in Dust staging and checked that they are not registered for our production Dust app either.

Is that expected?

**Pull request merged**